### ISSUE TITLE
__divmod and `clock-0.7.2' error with stack ghci on Win 10

### ISSUE THREAD

**Issue by AdamTSmith** [commercialhaskell/stack#4356]
Found when porting my Stack project to Win 10, and reduced to this minimal case: 
-	Create a new project with the default template, and put` hspec` in the package.yaml dependencies (no other changes).

Result (Stack Version 1.7.1):
```
Hask\try2DefTemp>stack ghci
[0mThe following GHC options are incompatible with GHCi and have not been passed to it: -threaded[0m
[0mConfiguring GHCi with the following packages: try2DefTemp[0m
[0mUsing main module: 1. Package `try2DefTemp' component exe:try2DefTemp-exe with main-is file: C:\Users\Cool\Hask\try2DefTemp\app\Main.hs[0m
GHCi, version 8.4.3: http://www.haskell.org/ghc/  :? for help
ghc.EXE:  | C:\sr\snapshots\0d969c47\lib\i386-windows-ghc-8.4.3\clock-0.7.2-Cf9UTsaN2AjEpBnoMpmgkU\HSclock-0.7.2-Cf9UTsaN2AjEpBnoMpmgkU.o: unknown symbol `___divmoddi4'
ghc.EXE: unable to load package `clock-0.7.2'
```

(The output has some Windows undisplayable character things by the way.)

The error only occurs with the command `stack ghci`. `stack build` and `stack test` work fine.

The closed GHC ticket #15094 indicates that the __divmod function and clock are used as Windows 32 bit does not provide the equivalent.  However I’m on 64 bit anyway so something is not quite right.

**Comment by mattaudesse** [comment]
Thanks @AdamTSmith - that's pretty strange.

I'm not really a Windows person myself so my ability to help you troubleshoot isn't that great, unfortunately. Hopefully someone else can chime in.

One platform-related question though: when you say you're porting to Windows 10, does that mean you've upgraded a system where your project (or the minimal case to reproduce) used to work, or are you trying to stand it up on an entirely different machine?

Also, do you have any `.ghci` configuration in place (either globally or local to the project) that might be relevant?

**Comment by AdamTSmith** [comment]
The only important bit is that the minimal case doesn't work on Win 10. 

(I'm not upgrading a system but trying to get my project to run on an entirely different machine that happens to have Win 10 on it, because it's faster than my Ubuntu laptop. That gave me the issue described, which I then experimented with to eventually pin down the problem as the use of the `hspec` package in a minimal case. It doesn't matter how I got there though.)

Stack was installed on the Win machine a while ago and I did `stack upgrade` to the current version. There's no `.ghci` file in the project directory (or actually on the entire machine). It does have a Winghci install on it though. Anyhow I thought Stack isolates and stops any interaction.

**Comment by mattaudesse** [comment]
That last part seems like a good clue. If this was a UNIX-like machine I'd be curious to know what `stack exec -- which ghci` reports back, but I'm actually not sure if that'll work in your case on Windows.

**Comment by AdamTSmith** [comment]
Works:
```
blah\Hask\try2DefTemp>stack exec -- which ghci
blah/AppData/Local/Programs/stack/i386-windows/ghc-8.4.3/bin/ghci
```
Do you mean the `Winghci` install is a clue? As I say I thought Stack isolated things by project (in default) so that you can have different versions for different projects, or from a main install. Or is that `ghc` only? (but that would be inconsistent and less useful).

**Comment by mattaudesse** [comment]
Yup, I thought investigating the `Winghci` angle might be a good place to start.

Isolation is definitely one of the biggest points of using `stack`, so I agree that's what we *should* be getting here as well.

Forgive the stupid question - I'm sure you already thought of this - but you definitely used the 64bit installer, right?
https://docs.haskellstack.org/en/stable/install_and_upgrade/#windows

**Comment by AdamTSmith** [comment]
@mattaudesse 
Your check was a good suggestion as I indeed had 32 bit Stack installed. (Can't believe I did that!) Confirmed by:
```
Prelude> :m + Foreign
Prelude Foreign> sizeOf (undefined :: Int)
4
Prelude Foreign> bitSize (undefined :: Int)
32
```

I then installed the 64 bit Win version and that got rid of the issue, as `stack ghci` now works on my minimal error project. In fact everything (ghci, build, exec, test) works on both the minimal project and my main project.

So this issue is a false alarm, unless the described behaviour is not what's wanted for a 32 bit install on a 64 bit Win machine. (Arguably it's not very informative for newish users.)
